src/prefect/orion/orchestration/core_policy.py

```diff
@@ -826,13 +829,14 @@ async def before_transition(
         self,
         initial_state: Optional[states.State],
         proposed_state: Optional[states.State],
-        context: TaskOrchestrationContext,
+        context: FlowOrchestrationContext,
     ) -> None:
-        await self.reject_transition(
-            state=states.Cancelled(),
-            reason=(
-                "Scheduled flows are transitioned immediately to Cancelled because there is no "
-                "infrastructure to shut down."
-            ),
-        )
-        return
+
+        if not context.run.infrastructure_pid:
+            await self.reject_transition(
+                state=states.Cancelled(),
+                reason=(
+                    "Scheduled flows are transitioned immediately to Cancelled because there is no "
+                    "infrastructure to shut down."
```

Comment:
This should probably just read: "Scheduled flow run has no infrastructure to terminate"